adapters/adhese/adhese.go

```diff
@@ -19,267 +13,205 @@ import (
 	"github.com/prebid/prebid-server/v3/util/jsonutil"
 )
 
-type AdheseAdapter struct {
+type adapter struct {
 	endpointTemplate *template.Template
 }
 
-func extractSlotParameter(parameters openrtb_ext.ExtImpAdhese) string {
-	return fmt.Sprintf("/sl%s-%s", url.PathEscape(parameters.Location), url.PathEscape(parameters.Format))
+type extTargetsAdhese map[string][]string
+type extImpAdheseWrapper map[string]extTargetsAdhese
+type wrappedBidExt map[string]map[string]string
+
+func makeSlot(params openrtb_ext.ExtImpAdhese) string {
+	return fmt.Sprintf("%s-%s", params.Location, params.Format)
 }
 
-func extractTargetParameters(parameters openrtb_ext.ExtImpAdhese) string {
-	if len(parameters.Keywords) == 0 {
-		return ""
+func (a *adapter) MakeRequests(
+	request *openrtb2.BidRequest,
+	requestInfo *adapters.ExtraRequestInfo,
+) (
+	[]*adapters.RequestData,
+	[]error,
+) {
+
+	imp := &request.Imp[0]
+
+	// Check if imp.Ext is empty before unmarshalling
+	if len(imp.Ext) == 0 {
+		return nil, []error{&errortypes.BadInput{Message: "imp.Ext is empty"}}
 	}
-	var parametersAsString = ""
-	var targetParsed map[string]interface{}
-	err := jsonutil.Unmarshal(parameters.Keywords, &targetParsed)
-	if err != nil {
-		return ""
+	var bidderExt adapters.ExtImpBidder
+	if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, []error{&errortypes.BadInput{Message: fmt.Sprintf("Error unmarshalling imp.ext: %v", err)}}
 	}
 
-	targetKeys := make([]string, 0, len(targetParsed))
-	for key := range targetParsed {
-		targetKeys = append(targetKeys, key)
+	// Check if bidderExt.Bidder is empty before unmarshalling
+	if len(bidderExt.Bidder) == 0 {
+		return nil, []error{&errortypes.BadInput{Message: "bidderExt.Bidder is empty"}}
 	}
-	sort.Strings(targetKeys)
-
-	for _, targetKey := range targetKeys {
-		var targetingValues = targetParsed[targetKey].([]interface{})
-		parametersAsString += "/" + url.PathEscape(targetKey)
-		for _, targetRawValKey := range targetingValues {
-			var targetValueParsed = targetRawValKey.(string)
-			parametersAsString += targetValueParsed + ";"
-		}
-		parametersAsString = strings.TrimRight(parametersAsString, ";")
+	var params openrtb_ext.ExtImpAdhese
+	if err := jsonutil.Unmarshal(bidderExt.Bidder, &params); err != nil {
+		return nil, []error{&errortypes.BadInput{Message: fmt.Sprintf("Error unmarshalling bidder ext: %v", err)}}
 	}
 
-	return parametersAsString
-}
-
-func extractGdprParameter(request *openrtb2.BidRequest) string {
-	if request.User != nil {
-		var extUser openrtb_ext.ExtUser
-		if err := jsonutil.Unmarshal(request.User.Ext, &extUser); err == nil {
-			return "/xt" + extUser.Consent
-		}
+	if params.Account == "" || params.Location == "" {
```

Comment:
Can you please double check? This if still present in the adapter code.
Also it doesn't match the properties setup in in adhere.json: "required": ["account", "location", "format"]

Reply:
i have removed the check, it is indeed already handled by the adhese.json